ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -49,9 +49,9 @@ let sync_required ~__context ~host =
         if cluster_rec.API.cluster_pool_auto_join
         then Some cluster_ref
         else None
-      | _ -> failwith "Internal error: More than one cluster_host object associated with this host"
+      | _ -> raise Api_errors.(Server_error (cluster_does_not_have_one_node, [ Ref.string_of host ]))
```

Comment:
I noticed that Xapi_cluster_host.destroy throws this error when the database returns multiple cluster hosts for one cluster, so I assume the error was created for that purpose. That said, the error isn't very clear so maybe it could be renamed to `cluster_should_only_have_one_cluster_host`?

Reply:
Please open a new CP ticket to discuss if to rename the error and to what, as this will probably require also XenCenter changes